Add documentation and cleanup CI workflow

parithosh opened the pull request.

jtraglia commented on README.md:
```diff
@@ -1,26 +1,55 @@
+# Setup Python environment
+python3 -m venv .venv
+source .venv/bin/activate
 pip install -r requirements.txt
```
I needed to install `wheel` and `setuptools` first here.
```suggestion
pip3 install wheel setuptools && pip3 install -r requirements.txt
```

jtraglia commented on README.md:
```diff
@@ -1,26 +1,55 @@
+
+1. Move into the scripts directory: `cd scripts`
+2. Download and unzip the test cases from the workflow run
+3. Install the requirements: `pip3 install ruamel.yaml==0.17.21 wheel setuptools && pip3 install -r requirements.txt`
```
It shouldn't be necessary to install `ruamel.yaml==0.17.21` here as it's listed in the requirements.txt file.
```suggestion
3. Install the requirements: `pip3 install wheel setuptools && pip3 install -r requirements.txt`
```

parithosh replied:
weirdly enough, both in the CI and on my builder machine I needed to explicitly install it again :/ I'm honestly not sure why :/

jtraglia commented on scripts/test_generated_output.py:
```diff
@@ -0,0 +1,173 @@
+                yield TestInfo(preset, fork, test_dir)
+
+
+def runt_tests_parallel(tests_dir, num_proc=settings.NUM_PROCESS):
```
Pretty sure this is a typo.
```suggestion
def run_tests_parallel(tests_dir, num_proc=settings.NUM_PROCESS):
```

jtraglia commented on scripts/test_generated_output.py:
```diff
@@ -0,0 +1,173 @@
+        help="directory with generated tests"
+    )
+    args = arg_parser.parse_args()
+    runt_tests_parallel(args.test_dir)
```
```suggestion
    run_tests_parallel(args.test_dir)
```

jtraglia reviewed (approved):
LGTM 👍 just some small nits!

jtraglia commented:
Also, I'd like to note that the tests don't appear to run in parallel for me. We can look into that later.

parithosh commented:
> Also, I'd like to note that the tests don't appear to run in parallel for me. We can look into that later.

Agreed that could be a future PR. Similarly, it would be nice to use `uv` instead of pip for the whole project.

ericsson49 commented:
>  Also, I'd like to note that the tests don't appear to run in parallel for me. We can look into that later.

The problem is that `GENERATOR_MODE = MODE_MULTIPROCESSING` should be appended to `../../core/pyspec/eth2spec/gen_helpers/gen_base/settings.py` before installing eth2spec.
But it's tricky with the current `requirements.txt`.

I think the best solution would be to implement a corresponding CLI switch in the `test_gen.py` script.

ericsson49 commented:
I've added `--fc-gen-multi-processing` CLI switch in [this commit](https://github.com/ericsson49/eth2.0-specs/commit/2eb1575634265860b792de249d542f94de7b937e)
So, it should work now without appending `GENERATOR_MODE = MODE_MULTIPROCESSING` to the `settings.py`

parithosh commented:
> I've added `--fc-gen-multi-processing` CLI switch in [this commit](https://github.com/ericsson49/eth2.0-specs/commit/2eb1575634265860b792de249d542f94de7b937e) So, it should work now without appending `GENERATOR_MODE = MODE_MULTIPROCESSING` to the `settings.py`

I've added a PR using it :)
https://github.com/ethereum/consensus-compliance-tests/pull/2
